### ISSUE TITLE
"rake db:migrate db:seed" elasticsearch error

### ISSUE THREAD

**Issue by fuyuasha** [square/shuttle#111]
Any suggestions (using elasticsearch 2.1.0_1)?

[ERROR] There has been an error when creating the index -- elasticsearch returned:
400 : {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"Mapping definition for [key_ids] has unsupported parameters:  [as : commits_keys.pluck(:key_id)]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [commit]: Mapping definition for [key_ids] has unsupported parameters:  [as : commits_keys.pluck(:key_id)]","caused_by":{"type":"mapper_parsing_exception","reason":"Mapping definition for [key_ids] has unsupported parameters:  [as : commits_keys.pluck(:key_id)]"}},"status":400}
[ERROR] There has been an error when creating the index -- elasticsearch returned:
400 : {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"Mapping definition for [index_in_section] has unsupported parameters:  [as : send(:key).index_in_section]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [translation]: Mapping definition for [index_in_section] has unsupported parameters:  [as : send(:key).index_in_section]","caused_by":{"type":"mapper_parsing_exception","reason":"Mapping definition for [index_in_section] has unsupported parameters:  [as : send(:key).index_in_section]"}},"status":400}

**Comment by RISCfuture** [comment]
I've seen this too. Seems to be related to a newer version of ElasticSearch. If you find out more let me know, but at some point we should probably switch to Chewy.

**Comment by fuyuasha** [comment]
Ah OK, uninstalled elasticsearch 2.1.0_1 and installed elasticsearch-0.20.6.tar.gz per https://github.com/karmi/retire - no more errors and things seem to be working.

Thanks

**Comment by yunussasmaz** [comment]
Yeah, it's a version problem. Elasticsearch 2.x deprecated a few things we are using. We are on Elasticsearch 1.2 at Square.

**Comment by fuyuasha** [comment]
OK ElasticSearch 1.2 - I'll use that - thanks for the update.

**Comment by fuyuasha** [comment]
Can I ask what version of rails is Square using for shuttle, I'm having some other problems and wonder if that might be the cause, I'm using:
rails (4.0.9)
rails-observers (0.1.2)

**Comment by yunussasmaz** [comment]
That's what we have, too: https://github.com/square/shuttle/blob/master/Gemfile.lock.

**Comment by fuyuasha** [comment]
Oh yep - I forget it's installed (and versioned) via the Gemfile.
